mdata/aggmetric.go

```diff
@@ -332,18 +252,12 @@ func (a *AggMetric) addAggregators(ts uint32, val float64) {
 
 // write a chunk to persistent storage. This should only be called while holding a.Lock()
 func (a *AggMetric) persist(pos int) {
-	if !cluster.Manager.IsPrimary() {
-		if LogLevel < 2 {
-			log.Debug("AM persist(): node is not primary, not saving chunk.")
-		}
-		return
-	}
-
-	pre := time.Now()
 	chunk := a.Chunks[pos]
+	pre := time.Now()
 
-	if chunk.Saved || chunk.Saving {
-		// this can happen if chunk was persisted by GC (stale) and then new data triggered another persist call
```

Comment:
was this scenario incorrect, or not important enough to be mentioned anymore in the new comment?

I suggest to change the comment to:
```
		// this can happen when:
		// * 2 primary nodes are saving chunks to Cassandra, A is ahead of B, B got a persist notification from A and then tries to save its chunk (would result in =)
		// * same, but B is lagging behind so much that A saved multiple chunks already (would result in > )
		// * same, but A crashed and B was promoted to primary
		// * chunk was persisted by GC (stale) and then new data triggered another persist call
```